torch/csrc/autograd/python_function.cpp

```diff
@@ -772,9 +812,21 @@ PyObject * THPFunction_do_backward(THPFunction *self, PyObject *args)
       THPUtils_invalidArguments(args, nullptr, "_do_backward", 1, "(tuple, bool)");
       return nullptr;
     }
-    THPUtils_assert(PyTuple_GET_SIZE(raw_grad_output) == self->cdata.num_inputs(),
+    auto cdata = self->cdata.lock();
+    // In obscure situations, cdata might be nullptr because it's expired.  THAT
```

Comment:
Uhhhh are you sure? If it's expired then how did we get to `do_backward`? Isn't it the `PyFunction` that calls this method?

Reply:
You can also just call `_do_backward` straight up. So the order of operations is, you save out the `grad_fn`, free the variables, and then call `grad_fn._do_backward`. It's dumb, not worth making it work better, but we do have an error check if this does happen :)